### PR TITLE
feat: [ACI-757] CredlyBadge credential creation

### PR DESCRIPTION
**Pull request by UvgenGen**
[ACI-757](https://youtrack.raccoongang.com/issue/ACI-757) CredlyBadge credential creation

### CODE REVIEW

File: credentials/apps/badges/signals/handlers.py [hunk 1]
```diff
@@ -51,9 +52,11 @@ def handle_badge_completion(sender, username, badge_template_uuid, **kwargs):  #
...
     """
 
-    # TODO: issue UserCredential
-    # TODO: ussue Credly badge
-    notify_badge_awarded(username, badge_template_uuid)
+    credential = BadgeTemplateIssuer.issued_credential_type.objects.get(uuid=badge_template_uuid)
+    BadgeTemplateIssuer().issue_credential(
+        credential,
+        username
+    )
```
Comment (wowkalucky) [inline credentials/apps/badges/signals/handlers.py:59]: I'm looking here for something like this:

```python
CredlyBadgeTemplateIssuer().issue(badge_template_id, username)
```
- so, BADGE_PROGRESS_COMPLETE signal is refactored from `uuid` to `id` usage;
- credential lookup (by `id`) is now an inner issuer responsibility (e.g. `_get_credential(credential_id)`;
- separate explicit issuer methods for issuing/revocation;

File: credentials/apps/badges/signals/handlers.py [hunk 2]
```diff
@@ -65,6 +68,9 @@ def handle_badge_regression(sender, username, badge_template_uuid, **kwargs):  #
     - badge template ID
     """
 
-    # TODO: change UserCredential's status to revoked
-    # TODO: revoke Credly badge
-    notify_badge_revoked(username, badge_template_uuid)
+    credential = BadgeTemplateIssuer.issued_credential_type.objects.get(uuid=badge_template_uuid)
+    BadgeTemplateIssuer().issue_credential(
```
Comment (wowkalucky) [inline credentials/apps/badges/signals/handlers.py:72]: ... and here:

```python
CredlyBadgeTemplateIssuer().revoke(badge_template_id, username)
```

**Review by wowkalucky — CHANGES_REQUESTED** [review]

File: credentials/apps/badges/services/issuers.py [hunk 3]
```diff
@@ -1,5 +1,138 @@
...
+
+class CredlyBadgeTemplateIssuer(BadgeTemplateIssuer):
+    """
+    Issues CredlyBadgeTemplate credentials to users.
+    """
+
+    issued_credential_type = CredlyBadgeTemplate
+    issued_user_credential_type = CredlyBadge
+
+    def notify_badge(self, username, badge_template_uuid, status):
```
[the remark this answers is not in the document]
Comment (UvgenGen) [reply credentials/apps/badges/services/issuers.py]: updated

File: credentials/apps/badges/signals/handlers.py (hunk 1 above)
Comment (UvgenGen) [reply credentials/apps/badges/signals/handlers.py:56]: updated

File: credentials/apps/badges/signals/handlers.py (hunk 2 above)
Comment (UvgenGen) [reply credentials/apps/badges/signals/handlers.py]: updated

**Review by wowkalucky — APPROVED** [review]

**Pull request merged**